client/user/src/app/(nonRoot)/apply/_components/Step/TermsAgreement.tsx

```
@@ -1,9 +1,58 @@
+    queryFn: () => fetchTerms(TermsType.SCAN),
+  })
+
+  const visibleTerms: Term[] =
+    termsData?.filter((term: Term) => term.isVisible) || []
+
+  useEffect(() => {
+    if (visibleTerms.length > 0) {
+      setTerms((prev: Record<number, boolean>) => {
+        const newTerms = visibleTerms.reduce<Record<number, boolean>>(
+          (acc, term) => ({
+            ...acc,
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)